src/compiler/utilities.ts

```diff
@@ -6007,6 +6050,29 @@ namespace ts {
         return node.kind === SyntaxKind.TypeReference || node.kind === SyntaxKind.ExpressionWithTypeArguments;
     }
 
+    /* @internal */
+    export function guessIndentation(lines: string[]) {
+        let indentation = Infinity;

```

Comment:
`Infinity` is definitely a double and not an SMI, so this will incorrectly optimize for double. A better value would be a sufficiently high UINT32 like `2 ** 30 - 1` (max SMI on x86, verified by running `node.exe --allow_natives_syntax -e "%IsSmi(2 ** 30 - 1)" -p` against an x86 copy of node).

Something like:
```ts
const MAX_SMI_X86 = 2 ** 30 - 1;
export function guessIndentation(lines: string[]) {
    let indentation = MAX_SMI_X86;
    ...
    return indentation === MAX_SMI_X86 ? undefined : indentation;
}
```